### PR TITLE
Apply the -runbundles+ decorator on the computed RunBundles when resolving

### PR DESCRIPTION
**Pull request by glimmerveen**
This PR should address issue #6364 . I applied the same mechanism used within [Project.getBundles](https://github.com/bndtools/bnd/blob/1c97750/biz.aQute.bndlib/src/aQute/bnd/build/Project.java#L634-L635) in the RunResolution's updateBundles function, just before the model is updated with the new RunBundles.

### CODE REVIEW

File: biz.aQute.resolve/src/biz/aQute/resolve/RunResolution.java [hunk 1]
```diff
@@ -208,7 +210,16 @@ public boolean updateBundles(BndEditModel model) {
 			newer = older;
 		}
 
-		model.setRunBundles(newer);
+		// Apply the -runbundles decorator on the computed RunBundles
```
Comment (pkriens) [inline biz.aQute.resolve/src/biz/aQute/resolve/RunResolution.java:213]: You have to change the check if the runbundles are identical before on line 197. The decoration might change the runbundles.

File: maven-plugins/bnd-resolver-maven-plugin/src/it/resolve-with-runstartlevel-and-runbundles-decorator/invoker.properties [hunk 2]
```diff
@@ -0,0 +1,7 @@
+invoker.goals=--no-transfer-progress package
```
Comment (pkriens) [inline maven-plugins/bnd-resolver-maven-plugin/src/it/resolve-with-runstartlevel-and-runbundles-decorator/invoker.properties:1]: Hmm. I am not too happy this testing takes place in the maven sub project. The functional change is in bnd so it should be tested in the normal bnd testing. The biz.aQute.resolve.RunResolutionTest in biz.aQute.resolve contains examples.

I know it is more comfortable to stay in your comfort zone but bnd is a shared project. If you change a function in bnd the test must take place as close as possible to the change.

**Review by pkriens — CHANGES_REQUESTED** [review]
Thanks for the fix. 

As said in a comment, I'd really appreciate if you move the tests to the RunResolutionTest to be closer to the original change.

File: maven-plugins/bnd-resolver-maven-plugin/src/it/resolve-with-runstartlevel-and-runbundles-decorator/invoker.properties (hunk 2 above)
Comment (glimmerveen) [reply maven-plugins/bnd-resolver-maven-plugin/src/it/resolve-with-runstartlevel-and-runbundles-decorator/invoker.properties:1]: That makes sense, I will look into adding test case(s) in RunResolutionTest. Do you want to retain these maven sub project test as well for this particular case, or just the tests within biz.aQute.resolve?

File: biz.aQute.resolve/src/biz/aQute/resolve/RunResolution.java (hunk 1 above)
Comment (glimmerveen) [reply biz.aQute.resolve/src/biz/aQute/resolve/RunResolution.java]: I have moved the application of decoration above the older/newer check.

**Comment by glimmerveen** [comment]
I have pushed a commit that should address the feedback. The only question I have is whether the tests in de maven sub module should be retained.

File: maven-plugins/bnd-resolver-maven-plugin/src/it/resolve-with-runstartlevel-and-runbundles-decorator/invoker.properties (hunk 2 above)
Comment (pkriens) [reply maven-plugins/bnd-resolver-maven-plugin/src/it/resolve-with-runstartlevel-and-runbundles-decorator/invoker.properties:1]: I think it is better to test it once in bnd. I consider ANY error that would be detected in a 'driver' (maven, gradle, eclipse, intellij) an error of the highest severity. We rarely have them fortunately. And in this case, the functionality is clearly in bnd, not the driver.

Comment (glimmerveen) [reply maven-plugins/bnd-resolver-maven-plugin/src/it/resolve-with-runstartlevel-and-runbundles-decorator/invoker.properties]: Ok, I have removed the integration tests from the maven sub module.

**Comment by pkriens** [comment]
thanks!

**Pull request merged**